native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceDispatcher.java

```diff
@@ -500,38 +503,35 @@ public static void dispatchOnText(WebSocketConnectionInfo connectionInfo, WebSoc
                                 bValue = StringUtils.fromString(stringAggregator.getAggregateString());
                                 break;
                             case TypeTags.XML_TAG:
-                                Object bXml = XmlUtils.parse(stringAggregator.getAggregateString());
-                                if (readOnly) {
-                                    bXml = CloneReadOnly.cloneReadOnly(bXml);
-                                }
-                                bValue = bXml;
+                                bValue = XmlUtils.parse(stringAggregator.getAggregateString());
                                 break;
                             case TypeTags.RECORD_TYPE_TAG:
-                                Object record = CloneWithType.convert(param, JsonUtils.parse(
+                                bValue = CloneWithType.convert(param, JsonUtils.parse(
                                 stringAggregator.getAggregateString()));
-                                if (record instanceof BError) {
-                                    sendDataBindingError(webSocketConnection, ((BError) record).getMessage());
-                                    return;
-                                }
-                                if (readOnly) {
-                                    record = CloneReadOnly.cloneReadOnly(record);
+                                break;
+                            case TypeTags.UNION_TAG:
+                                if (WebSocketUtil.hasStringType(param)) {
+                                    bValue = CloneWithType.convert(param,
+                                            StringUtils.fromString(stringAggregator.getAggregateString()));
+                                } else {
+                                    bValue = FromJsonStringWithType.fromJsonStringWithType(StringUtils.fromString(
```

Comment:
we can send the else block to default case right?

Reply:
Changed the code to fall through.